bench/benches/packets.rs

```diff
@@ -1,20 +1,51 @@
 use criterion::{black_box, criterion_group, criterion_main, Criterion};
 use nb2::packets::ip::v4::Ipv4;
 use nb2::packets::{Ethernet, Packet, Udp};
-use nb2::testils::byte_arrays::UDP_PACKET;
+use nb2::testils::proptest::*;
+use nb2::testils::{PacketExt, StrategyValGen};
 use nb2::Mbuf;
+use std::time::{Duration, Instant};
 
-fn parse_ipv4_packet() {
-    let packet = Mbuf::from_bytes(&UDP_PACKET).unwrap();
-    let ethernet = packet.parse::<Ethernet>().unwrap();
+fn parse_udp_packet(mbuf: Mbuf) {
+    let ethernet = mbuf.parse::<Ethernet>().unwrap();
     let ipv4 = ethernet.parse::<Ipv4>().unwrap();
     ipv4.parse::<Udp<Ipv4>>().unwrap();
 }
 
+fn deparse_udp_packet(udp: Udp<Ipv4>) {
+    let d_ipv4 = udp.deparse();
+    let d_eth = d_ipv4.deparse();
+    d_eth.deparse();
+}
+
 #[nb2::bench]
 fn parse_benchmark(c: &mut Criterion) {
-    c.bench_function("packets::parse_ipv4_packet", |b| {
-        b.iter(|| black_box(parse_ipv4_packet()));
+    c.bench_function("packets::parse_udp_packet", move |b| {
+        b.iter_custom(|iters| {
+            let mut gen = StrategyValGen::new();
+            let mut total_elapsed = Duration::from_secs(0);
+            for _i in 0..iters {
+                let packet_mbuf = gen.generate(v4_udp());
+                let start = Instant::now();
+                black_box(parse_udp_packet(packet_mbuf));
+                total_elapsed += start.elapsed()
```

Comment:
Oh I see ok you are measuring each iteration and then adding the elapsed together. Hmm, I think this is very prune to the precision problems. if you have something that's tiny per iteration which a lot of times these things can be, this will not give you a good precise measurement at the end.